### PR TITLE
File endpoint

### PR DESCRIPTION
**Pull request by MDunitz**
## Need
- allow users to simply retrieve a list of bundle fqids associated with a given file uuid
- allow users to retrieve a list of file fqids associated with a given schema type and version

## Approach
- create file/file_uuid/bundles endpoint which returns a list of bundle fqids
- create file/schema_type endpoint which returns a list of file fqids

## Tests
- unit tests
- integration tests

### CODE REVIEW

File: dcpquery/api/files/schema_type.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import requests
+from flask import Response
+
+from dcpquery import config
+from dcpquery.db.models import File, DCPMetadataSchemaType
+
+
+def get(schema_type, version=None):
```
Comment (maniarathi) [inline dcpquery/api/files/schema_type.py]: Could I request for docstrings throughout this PR actually please?

File: dcpquery-api.yml [hunk 2]
```diff
@@ -148,6 +148,85 @@ paths:
...
+            in: path
+            description: Job identifier in RFC4122-compliant UUID format
+            required: true
+            schema:
+              type: string
+              pattern: "[A-Za-z0-9]{8}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{12}"
+          - name: check_events
+            in: query
+            description: >
+              If set, will check for any recent dss events and check if the file is a part of any unloaded bundles.
```
Comment (maniarathi) [inline dcpquery-api.yml]: Maybe there's a slightly better word that "unloaded"? Uncommitted? Unindexed?

File: dcpquery-api.yml [hunk 3]
```diff
@@ -148,6 +148,85 @@ paths:
...
+        description: >
+          Given a schema type, and major and minor versions, returns a list of file_fqids that have a
+          json body based on the schema defined by the given values
+        parameters:
+          - name: schema_type
+            in: path
+            description: Job identifier in RFC4122-compliant UUID format
+            required: true
+            schema:
+              type: string
```
Comment (maniarathi) [inline dcpquery-api.yml]: Would you want to specify a pattern here as well?

File: dcpquery-api.yml [hunk 4]
```diff
@@ -148,6 +148,85 @@ paths:
...
+          - name: version
+            in: query
+            description: >
+              Set to the major and minor versions of the schema
+            schema:
+              type: string
```
Comment (maniarathi) [inline dcpquery-api.yml]: Could also benefit from a pattern here as well to drive home the point that the patch version isn't specified.

File: dcpquery/api/files/bundle.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+def get(file_uuid, check_events):
+    """
+    Function called by /files/{file_uuid}/bundles endpoint
+    :param file_uuid: UUID of file of interest
+    :param check_events: Boolean to flag if it is necessary to check for recent bundle events from the dss
+    :return: an object containing the file_uuid and a list of bundle_fqids that file belongs to
+    """
+    if check_events:
+        pass
+        # TODO when dss event replay endpoint available, set up cron job and check for missed events
```
Comment (maniarathi) [inline dcpquery/api/files/bundle.py]: Just a question: is there an open ticket to track this? Could potentially reference it here.

File: dcpquery/db/models/__init__.py [hunk 6]
```diff
@@ -94,6 +99,13 @@ class File(DCPQueryModelHelper, SQLAlchemyBase):
     def select_file(cls, file_fqid):
         return config.db_session.query(cls).filter(cls.fqid == file_fqid).one_or_none()
 
+    @classmethod
+    def select_files_for_uuid(cls, file_uuid):
+        files = config.db_session.query(cls).filter(cls.uuid == file_uuid).all()
+        if len(files) == 0:
```
Comment (maniarathi) [inline dcpquery/db/models/__init__.py]: I'm a little confused by this conditional. Shouldn't there only ever be one file for one file_uuid? If there is more than one, that seems like an error as well?

This also goes for the `get` function which uses `select_files_for_uuid`. It seems like you iterate through a list of files that are associated with this uuid but it seems like that wouldn't really make sense?

Comment (MDunitz) [reply dcpquery/db/models/__init__.py]: Files are specified by uuid and version (or fqid). There can be many files associated with the same uuid. The filter function will return an empty array if no matches are found.

File: dcpquery-api.yml (hunk 2 above)
Comment (MDunitz) [reply dcpquery-api.yml]: 👍

File: dcpquery-api.yml (hunk 4 above)
Comment (MDunitz) [reply dcpquery-api.yml]: 👍

File: dcpquery-api.yml (hunk 3 above)
Comment (MDunitz) [reply dcpquery-api.yml]: 👍

**Pull request merged**